brownie/src/index.ts

```diff
@@ -0,0 +1,162 @@
+import pg, { PoolConfig } from "pg"
+
+import { logger } from "./logger.js"
+import { readParams } from "./bootstrap.js"
+import { Config } from "./config.js"
+import { ResourceStatus, evaluateResource } from "./utils.js"
+import Kafkajs from "kafkajs"
+import type { KafkaConfig as KafkaJsConfig, SASLOptions } from "kafkajs"
+
+const main = async () => {
+  const config: Config = readParams()
+  logger.info("main(), Parsed configuration: \n%s", JSON.stringify({ ...config, timestampPattern: config.timestampPattern.toString() }, null, 2))
+
+  logger.info("Cleaning old databases...")
+  const cleanedDbsCount = await cleanOldDatabases(config)
+  if (cleanedDbsCount > 0) {
+    logger.info("Dropped %s database%s", cleanedDbsCount, cleanedDbsCount > 1 ? "s" : "")
+  } else {
+    logger.info("There are no databases to clean")
+  }
+  logger.info("\n\n")
+
+  logger.info("Cleaning kafka topics...")
+  const cleanedTopicsCount = await cleanTopics(config)
+  if (cleanedTopicsCount > 0) {
+    logger.info("Deleted %s topics%s", cleanedTopicsCount, cleanedTopicsCount > 1 ? "s" : "")
+  } else {
+    logger.info("There are no topics to delete")
+  }
+}
+
+const cleanOldDatabases = async (config: Config): Promise<number> => {
+  let cleanedCount = 0
+  const pgClient = new pg.Client({
+    user: config.pg.username,
+    database: config.pg.database,
+    password: config.pg.password,
+    port: config.pg.port,
+    host: config.pg.host,
+    min: 1
+  } as PoolConfig)
+
+  await pgClient.connect()
+  try {
+    const result = await pgClient.query({
+      name: "select-databases",
+      text: "SELECT d.datname as database FROM pg_catalog.pg_database d WHERE d.datname like $1 ORDER BY d.datname",
+      values: ["%pit%"]
+    })
+    for (let row of result?.rows) {
+      const dbname = row.database
+
+      try {
+        const status = evaluateResource(dbname, config.timestampPattern, new Date(), config.retentionMinutes)
```

Comment:
Although I would like to avoid the timestamp as part of the kafka topic for the cleanup to work, unlike postgres suggestion I made above, it seems harder with Kafka topics.

Couple of possibilities which we could maybe explore in the future is:- 
1. Spawn a child process which 
 - would via a bash script get all the topics from either Kafka broker or zookeeper
-  Get the creation time from zookeeper using stats
-  Delete the topics that are no longer relevant.

2. Keep track of topics created with the date of creation when they are created and use that as part of clean up. 

As neither are trivial, I am throwing some ideas for future.